### PR TITLE
Migrate from the `command-exists-promise` dependency to `command-exists`

### PR DESCRIPTION
**Pull request by dhenneke**
The scaffolder uses the [`command-exists-promise`](https://www.npmjs.com/package/command-exists-promise) dependency which is a fork of [`command-exists`](https://www.npmjs.com/package/command-exists). While the latter one has >100 Stars and >1.4 million downloads, the former has only 7 stars and 14000 downloads.

I don't see a real reason why one would use the fork since the original one also has native support for the promise api. It is also already used in the `check-docs-quality.js` file. Thus, I would propose to switch to the more common package `command-exists` as I find it a good practice to prefer popular packages.

We would like to change it because FOSSA complains with a lot of strange issues and we hope that we can get rid of them by switching to the other package. However, this might also be an error on FOSSA side. There also seem to be issues in the FOSSA integration in this project, though we can't see any details without being part of the (what seems to be the) CNCF FOSSA account: https://app.fossa.com/projects/custom%2B162%2Fbackstage/refs/branch/master/838f4c779a982b1c1b47231b0903cb81b30e23d7/browse/dependencies/?search=command-exists

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import path from 'path';
 import { runCommand } from './helpers';
 import { TemplaterBase, TemplaterRunOptions } from './types';
 
-const commandExists = require('command-exists-promise');
+const commandExists = require('command-exists');
```
Comment (Fox32) [inline plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts:24]: Doesn't `command-exists` provide typings that you could use here?

Comment (Fox32) [reply plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts:24]: See https://www.npmjs.com/package/@types/command-exists

**Comment by Fox32** [comment]
`command-exists` also seems to be more recent, so another point for not using the fork 😉

File: plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts (hunk 1 above)
Comment (dhenneke) [reply plugins/scaffolder-backend/src/scaffolder/stages/templater/cookiecutter.ts]: Updated it 👍

**Review by freben — APPROVED** [review]

**Pull request merged**